src/persist-client/src/lib.rs

```diff
@@ -54,6 +54,25 @@ pub(crate) mod r#impl {
 // TODOs
 // - Decide if the inner and outer error of the two-level errors should be
 //   swapped.
+// - Split LocationError into definitely failed vs maybe failed variants.
```

Comment:
I'm becoming increasingly confidant that the simple LocationError/InvalidUsage split isn't quite right. there's a number of places here where InvalidUsage doesn't really sound appropriate but ended up needing to be used. largely, I feel like we need to punt on this while things shake out for a bit

Reply:
the more I think about it, the more I'm starting to wonder if indeed kyle's definite/indefinite split is the only structure we need. concretely: maybe we could get away with replacing InvalidUsage+LocationError with something like

```
struct Maybe {
    pub inner: anyhow::Error,
}
struct Failed {
    pub inner: anyhow::Error,
}
```

Reply:
can it be `Maybe🤷‍♂️`?

Reply:
More seriously, though: should there be a definite/indefinite. It seems like an `InvalidUsage` error is always definite because we do know that someone did something wrong.

Another data point: my PR for adding `compare_and_append()` (which will conflict big-time with this) has another layer of `Result`, which I don't like. But the result of a failed cas operation seems to be neither a usage error nor a "location" error.

Also: I thought maybe `ExternalError` could work for what we call `LocationError` right now.